client/lib/cart-values/cart-items.js

```diff
@@ -696,6 +696,16 @@ function isNextDomainFree( cart ) {
 	return !! ( cart && cart.next_domain_is_free );
 }
 
+function isDomainBeingUsedForPlan( cart, domain ) {
+	if ( cart && domain && hasPlan( cart ) ) {
+		const domainProducts = getDomainRegistrations( cart ).concat( getDomainMappings( cart ) ) || [],
```

Comment:
The `|| []` part doesn't make sense to me - even if `getDomainRegistrations` and `getDomainMappings` removed empty lists, `[]` is still a truthy value, by JS standards :)